maven-scm-providers/maven-scm-providers-git/maven-scm-provider-gitexe/src/test/java/org/apache/maven/scm/provider/git/gitexe/command/changelog/GitChangeLogConsumerTest.java

```diff
@@ -172,4 +173,72 @@ public void testConsumer2()
         assertTrue( cf.getRevision() != null && cf.getRevision().length() > 0 );
     }
 
+    public void testGitLogConsumer3()
+            throws Exception
+    {
+        GitChangeLogConsumer consumer = new GitChangeLogConsumer( new DefaultLog(), null );
+
+        File f = getTestFile( "/src/test/resources/git/changelog/gitlog3.gitlog" );
+
+        ConsumerUtils.consumeFile( f, consumer );
+
+        List<ChangeSet> modifications = consumer.getModifications();
+
+        assertEquals( 10, modifications.size() );
+
+        SimpleDateFormat sdf = new SimpleDateFormat( "yyyy-MM-dd HH:mm:ss Z" );
+        sdf.setTimeZone( TimeZone.getTimeZone( "GMT" ) );
```

Comment:
This is error-prone and will likely fail. Git, unfortunately, stores timestamp and offset. You never know in which offset the output will be because it depends on the author's/committer's system offset. Ideally, the pattern autorecognizes the offset.

WDYT?

Reply:
The file contains a timestamp in a timezone which is parsed into an instance of `Date`. All this does is format this parsed date into a String so it is an easier to read assertion. How is that error prone?

Reply:
My bad, I was thinking about the inverse operation. Day was too long today.

Reply:
Np.